src/iptvsimple/data/EpgEntry.cpp

```diff
@@ -220,15 +220,20 @@ bool EpgEntry::UpdateFrom(const xml_node& channelNode, const std::string& id,
 
   m_genreString = GetJoinedNodeValues(channelNode, "category");
 
-  const std::string dateString = GetNodeValue(channelNode, "date");
+  std::string dateString = GetNodeValue(channelNode, "date");
   if (!dateString.empty())
   {
     static const std::regex dateRegex("^[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]");
     if (std::regex_match(dateString, dateRegex))
     {
-      m_firstAired = ParseAsW3CDateString(dateString);
-      // Always compare to the raw string date value to avoid timezone issues
-      if (m_firstAired == ParseAsW3CDateString(strStart))
+      dateString = dateString.substr(0,8) + strStart.substr(8);
+      long long tmpDate = ParseDateTime(dateString);
+      // Windows localtime_s doesnt support negative time_t
+      if (tmpDate < 0)
+        m_firstAired = ParseAsW3CDateString(dateString);
+      else
+        m_firstAired = ParseAsW3CDateString(static_cast<time_t>(tmpDate));
+      if (m_firstAired == ParseAsW3CDateString(m_startTime))
```

Comment:
You will still want to compare this is strStart right?

Reply:
Or actually the check needs to depend on which branch it takes for m_firstAired

Reply:
nope, as that wouldn't be in "local time". 
We want to now compare both local times
only time it won't work is for dates before 1970
we wont have m_startTime going back that far anyway

Reply:
if you wanted it 100% even showing "new" for epg entries from 1960, you would do
```
      if (tmpDate < 0) {
        m_firstAired = ParseAsW3CDateString(dateString);
        if (m_firstAired == ParseAsW3CDateString(strStart))
            m_new = true;
        }
      else{
        m_firstAired = ParseAsW3CDateString(static_cast<time_t>(tmpDate));
        if (m_firstAired == ParseAsW3CDateString(m_startTime))
            m_new = true;
        }
```

Reply:
Yes, this is what I meant. But it could be shorter:

```
      if (tmpDate < 0) {
        m_firstAired = ParseAsW3CDateString(dateString);
        m_new = m_firstAired == ParseAsW3CDateString(strStart);
            m_new = true;
        }
      else{
        m_firstAired = ParseAsW3CDateString(static_cast<time_t>(tmpDate));
        m_new = m_firstAired == ParseAsW3CDateString(m_startTime);
        }
```

Reply:
oooo, thats pretty :)
done!
(even though a m_firstAired of before 1970 would be odd!)